prov/util/src/util_mr_map.c

```diff
@@ -108,22 +108,32 @@ int ofi_mr_map_verify(struct ofi_mr_map *map, uintptr_t *io_addr,
 	void *addr;
 
 	node = ofi_rbmap_find(map->rbtree, &key);
-	if (!node)
-		return -FI_EINVAL;
+	if (!node) {
+                FI_DBG(map->prov, FI_LOG_MR, "verify_addr: unknown key: %" PRIu64 "\n",
+                  key);
+	        return -FI_EINVAL;
+        }
 
 	attr = node->data;
 	assert(attr);
 
 	if ((access & attr->access) != access) {
-		FI_DBG(map->prov, FI_LOG_MR, "verify_addr: invalid access\n");
+                FI_DBG(map->prov, FI_LOG_MR, "verify_addr: invalid access: %s and %s\n",
+                  fi_tostr(&attr->access, FI_TYPE_MR_MODE),
+                  fi_tostr(&access, FI_TYPE_MR_MODE));
```

Comment:
fi_tostr() uses an internal character array.  Calling it twice will replace the contents of the array.  Either use 2 separate prints, one per fi_tostr() call, print the flags in hex, or use fi_tostr_r() and pass in buffers.  The easiest but most useful is 2 prints.

The text here should indicate "permitted access: xxx, requested access: yyy", to make it clear which value is which.